rd-kt/rd-framework/src/test/kotlin/com/jetbrains/rd/framework/test/cases/RdAsyncTaskTest2.kt

```diff
@@ -0,0 +1,128 @@
+
+    @Test
+    fun defaultResponseScheduler() {
+        val entity_id = 1
+
+        val callsite = RdCall<Int, String>().static(entity_id).apply { async = true }
+        val endpoint = RdCall<Int, String>(null) { x -> throw Exception() }.static(entity_id)
+
+        runBlocking(clientScheduler.asCoroutineDispatcher) {
+            clientProtocol.bindStatic(callsite, "client")
+        }
+
+        runBlocking(serverScheduler.asCoroutineDispatcher) {
+            serverProtocol.bindStatic(endpoint, "server")
+        }
+
+        endpoint.setSuspend { lifetime, arg ->
+            serverScheduler.assertThread()
+            yield()
+            serverScheduler.assertThread()
+            arg.toString()
+        }
+
+
+        runBlocking(clientScheduler.asCoroutineDispatcher) {
+            runBlocking {
+                withTimeout(5000L) {
+                    val result = callsite.startSuspending(5)
+                    Assertions.assertEquals("5", result)
+                }
+            }
+        }
+    }
+
+    @Test
+    fun defaultResponseSchedulerWithBindableResult() {
+        val entity_id = 1
+
+        val callsite = RdCall(FrameworkMarshallers.Void, RdSignal.Companion as ISerializer<RdSignal<Int>>).static(entity_id).apply { async = true }
+        val endpoint = RdCall(FrameworkMarshallers.Void, RdSignal.Companion as ISerializer<RdSignal<Int>>).static(entity_id)
+
+        runBlocking(clientScheduler.asCoroutineDispatcher) {
+            clientProtocol.bindStatic(callsite, "client")
+        }
+
+        runBlocking(serverScheduler.asCoroutineDispatcher) {
+            serverProtocol.bindStatic(endpoint, "server")
+        }
+
+        val signal = RdSignal<Int>()
+
+        val list = mutableListOf<Int>()
+
+        Lifetime.using { adviseLifetime ->
+
+            signal.advise(adviseLifetime) { value ->
+                serverScheduler.assertThread()
+                list.add(value)
+            }
+
+            endpoint.setSuspend { lifetime, arg ->
+                serverScheduler.assertThread()
+                yield()
+                serverScheduler.assertThread()
+                signal
+            }
+
+
+            runBlocking(clientScheduler.asCoroutineDispatcher) {
+                val def = LifetimeDefinition()
+                val lifetime = def.lifetime
+
+                val result = runBlocking {
+                    withTimeout(5000L) {
+                        callsite.startSuspending(lifetime, Unit)
+                    }
+                }
+
+                result.fire(1)
+                withContext(serverScheduler) {
+                    assertEquals(1, list.single())
+                }
+
+                result.fire(2)
+                withContext(serverScheduler) {
+                    assertEquals(1, list[0])
+                    assertEquals(2, list[1])
+                    assertEquals(2, list.size)
+                }
+
+                def.terminate()
```

Comment:
Please use `try/finally` to not create a lifetime leak even in case of a test failure, and always terminate this lifetime.

Reply:
It doesn't matter in this case, but I will rewrite this code.